cupy/cuda/__init__.py

```diff
@@ -51,15 +51,7 @@ def __init__(self, name):
 except ImportError:
     thrust = _UnavailableModule('cupy.cuda.thrust')
 
-try:
-    from cupy.cuda import nccl  # NOQA
-except ImportError:
-    nccl = _UnavailableModule('cupy.cuda.nccl')
-
-try:
-    from cupy_backends.cuda.libs import cutensor
-except ImportError:
-    cutensor = _UnavailableModule('cupy.cuda.cutensor')
+from cupy.cuda import cutensor  # NOQA
```

Comment:
(sorry, repost to make the discussion appear in the PR.)

Looks like there are a number of places in the codebase expecting to do `if cupy.cuda.cutensor.available: ...`, so I had to import `cutensor` here, making the treatment a bit asymmetric w.r.t. other two optional dependencies (cudnn and nccl). @kmaehashi WDYT?

Reply:
How about rewriting `if cupy.cuda.cutensor.available:` with this?

```py
try:
    import cupy_backends.cuda.libs.cutensor
except ImportError:
    pass
```

Reply:
Wouldn't that make the `available` flag entirely useless?

Reply:
Perhaps this is better?
```python
from cupy.cuda import cutensor
if cutensor.available:
    ...
```

Reply:
I think `available` flag is still useful for end-users. e.g, `if cupy.cuda.cutensor.available: import cupy.cutensor`

Reply:
thanks, done in a18fe9e.

Reply:
So, without importing the wrapper modules in `cupy/cuda/__init__.py`, end users would see `AttributeError` when doing `if cupy.cuda.cutensor.available: ...`. I think they still need to be imported...

Reply:
Sorry, @kmaehashi I think I confused myself 😂 Let me summarize:
1. If `nccl`/`cutensor` is imported in `cupy/cuda/__init__.py`:
    - We restore the backward compatibility (as nothing is lost), in particular users can still write `if cupy.cuda.cutensor.available: ...`
    - but we bring in warnings at cupy import time if the library is not installed
2. If `nccl`/`cutensor` is NOT imported in `cupy/cuda/__init__.py`:
    - We lose backward compatibility, so user codes like `if cupy.cuda.cutensor.available: ...` will stop working
    - warnings are not raised until the module is actually needed

I suppose Option No.2 is what you are looking for?

Reply:
Yes, Option 2 is what I meant.  User code `import cupy.cuda; if cupy.cuda.cutensor.available` will stop working, and users need to rewrite it as `import cupy.cuda.cutensor; if cupy.cuda.cutensor.available`. `import cupy.cuda.cutensor` should generate a warning, but `import cupy.cuda` should not.

Reply:
Thanks for bearing with me @kmaehashi. Will fix the other PR shortly.